sql/stmtdiagnostics: TestChangePollInterval failed

[(sql/stmtdiagnostics).TestChangePollInterval failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2786002&tab=buildLog) on [master@597e4a8c487e3c23d64885563d608a692b59055c](https://github.com/cockroachdb/cockroach/commits/597e4a8c487e3c23d64885563d608a692b59055c):

```
I210317 07:46:17.365911 4566 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 223 ={"Timestamp":1615967177363067087,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"3dfbfa15-a35d-415c-a7d7-eccb33839f0b"}
I210317 07:46:17.369968 4166 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 224 ={"Timestamp":1615967177367703431,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210317 07:46:17.372657 4576 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 225 ={"Timestamp":1615967177370176064,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210317 07:46:17.389150 3152 server/server_sql.go:840  [n1] 226  done ensuring all necessary startup migrations have run
I210317 07:46:17.389222 3152 1@server/server.go:2061  [n1] 227  serving sql connections
I210317 07:46:17.389252 4840 jobs/job_scheduler.go:360  [n1] 228  waiting 2m0s before scheduled jobs daemon start
    statement_diagnostics_test.go:209: 
        	Error Trace:	statement_diagnostics_test.go:209
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestChangePollInterval
W210317 07:46:17.389765 4720 kv/kvserver/intentresolver/intent_resolver.go:758  [-] 229  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W210317 07:46:17.389862 3696 jobs/registry.go:714  [-] 230  canceling all adopted jobs due to stopper quiescing
W210317 07:46:17.389976 3695 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 231  exiting heartbeat loop
W210317 07:46:17.389960 4823 kv/txn.go:635  [n1,intExec=show-version] 233  failure aborting transaction: node unavailable; try another peer; abort caused by: read-setting: context canceled
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232  unable to read orphaned leases: read orphaned leases: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +(1) attached stack trace
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  -- stack trace:
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:658
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next.func1
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:273
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*rowsIterator).Next
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:313
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).queryInternalBuffered
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:414
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryBufferedEx
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:394
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).QueryBuffered
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/internal.go:376
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).DeleteOrphanedLeases.func1.1
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/lease.go:2067
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/util/retry.WithMaxAttempts
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:172
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).DeleteOrphanedLeases.func1
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/sql/catalog/lease/lease.go:2065
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:351
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | runtime.goexit
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +Wraps: (2) read orphaned leases
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +Wraps: (3) failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  |
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | (opaque error leaf)
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +  | type name: github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/kvcoord.sendError
W210317 07:46:17.389918 4839 sql/catalog/lease/lease.go:2072  [-] 232 +Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueLeaf
W210317 07:46:17.390099 4823 kv/txn.go:635  [n1,intExec=show-version] 234  failure aborting transaction: node unavailable; try another peer; abort caused by: query execution canceled
I210317 07:46:17.390221 4842 server/auto_upgrade.go:52  [n1] 235  failed attempt to upgrade cluster version, error: show-version: context canceled
--- FAIL: TestChangePollInterval (10.29s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestChangePollInterval PKG=./pkg/sql/stmtdiagnostics TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestChangePollInterval.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/stmtdiagnostics/BUILD.bazel
+++ b/pkg/sql/stmtdiagnostics/BUILD.bazel
@@ -42,6 +42,7 @@ go_test(
         "//pkg/security",
         "//pkg/security/securitytest",
         "//pkg/server",
+        "//pkg/settings/cluster",
         "//pkg/sql",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",

--- a/pkg/sql/stmtdiagnostics/stament_diagnostics_helpers_test.go
+++ b/pkg/sql/stmtdiagnostics/stament_diagnostics_helpers_test.go
@@ -18,3 +18,6 @@ func (r *Registry) InsertRequestInternal(ctx context.Context, fprint string) (in
 	id, err := r.insertRequestInternal(ctx, fprint)
 	return int64(id), err
 }
+
+// PollingInterval is exposed to override in tests.
+var PollingInterval = pollingInterval

--- a/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
+++ b/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
@@ -21,7 +21,9 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/stmtdiagnostics"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -184,7 +186,13 @@ func TestChangePollInterval(t *testing.T) {
 		})
 		return seen
 	}
+	settings := cluster.MakeTestingClusterSettings()
+
+	// Set an extremely long initial polling interval to not hit flakes due to
+	// server startup taking more than 10s.
+	stmtdiagnostics.PollingInterval.Override(&settings.SV, time.Hour)
 	args := base.TestServerArgs{
+		Settings: settings,
 		Knobs: base.TestingKnobs{
 			Store: &kvserver.StoreTestingKnobs{
 				TestingRequestFilter: func(ctx context.Context, request roachpb.BatchRequest) *roachpb.Error {

